.github/workflows/hermetic_library_generation.yaml

```diff
@@ -14,14 +12,59 @@ jobs:
     steps:
     - uses: actions/checkout@v4
       with:
-        ref: ${{ github.head_ref }}
+        fetch-depth: 0
+    - name: checkout the target branch
+      shell: bash
+      run: |
+        git checkout ${{ github.base_ref }}
+    - name: checkout the current branch
+      shell: bash
+      run: |
+        git checkout ${{ github.head_ref }}
     - name: get baseline generation config
-      uses: actions/checkout@v4
-      with:
-        ref: ${{ github.base_ref }}
-        path: baseline
-        sparse-checkout: generation_config.yaml
-    - name: diff
       shell: bash
       run: |
-        diff generation_config.yaml baseline/generation_config.yaml
+        git show ${{ github.base_ref }}:generation_config.yaml > baseline_generation_config.yaml
+    - name: show configuration diff
```

Comment:
Can we extract most of the logics here to a shell script so that we can call the script directly here? That way we can easily call it in local and possibly in a non-Github environment in the future.
Ideally, we should try to put this logic into Docker container as well, so that this logic could be reused by handwritten libraries, otherwise we would need to duplicate this in every handwritten repos. Bake it into Docker can be a future enhancement, but I think we should at least extract it to a separate script.

Reply:
Move command lines to a script.